lib/evmone/eof.cpp

```diff
@@ -349,40 +349,41 @@ bool validate_rjump_destinations(bytes_view code) noexcept
 std::variant<EOFValidationError, int32_t> validate_max_stack_height(
     bytes_view code, size_t func_index, const std::vector<EOFCodeType>& code_types)
 {
-    assert(!code.empty());
-
     // Special values used for detecting errors.
     static constexpr int32_t LOC_UNVISITED = -1;  // Unvisited byte.
-    static constexpr int32_t LOC_IMMEDIATE = -2;  // Immediate byte.
+
+    struct StackHeightRange
+    {
+        int32_t min = 0;
```

Comment:
It looks it is better to init them with "unvisited". You can also move the constant inside the struct.

```suggestion
        int32_t min = LOC_UNVISITED;
```

Reply:
Changed the default. Decided not to move the constant, to avoid `StackHeightRange::LOC_UNVISITED` below.